list/directory/list.js

```diff
@@ -0,0 +1,479 @@
+     * @returns {string} initial selected data from grouped array
+     * @private
+     */
+    _getInitialSelectedLetter(data){
+        let initialLetter;
+        if((initialLetter = keys(data)[0]) !== undefined){
+            return initialLetter;
+        }
+    },
+
+    /**
+     * Retrieve available letters from data.
+     * Hint: The grouped data is already sorted by letter.
+     * @param {object} grouped data. exemple: {A: Array[], B: Array[], C: Array[], ..., Z: Array[]}
+     * @returns {string} a string of available letters
+     * @private
+     */
+    _getAvailableLetters(data){
+        return keys(data).join('');
+    },
+
+    /**
+     * Handle letter click event.
+     * if showAll is true, scroll to letter group. Otherwise, show only group related to selected letter
+     * @param {string} selected letter
+     * @private
+     */
+    _onLetterSelected(letter){
+        if(this.props.showAll){//display all data
+            let groupView = document.querySelector(`[data-directory-group="${letter}"]`);
+            groupView.scrollIntoView();
+        }else{//display only one group
+            let dataToUse = this._getDataToUse();
+            let availableLetters = this._getAvailableLetters(dataToUse);
+            let groupToDisplay = this._getGroupFromData(dataToUse, letter);
+            this.setState({data: groupToDisplay, currentSelectedLetter: letter, availableLetters: availableLetters});
+        }
+
+    },
+
+    /**
+     * Filter results by searching text in data.
+     * @private
+     */
+    _filterResults(){
+        this.searchText = this.quickSearchDOMNode.value.trim();
+        let newState = this._getState();
+
+        this.setState(
+            {   data: newState.data,
+                currentSelectedLetter: newState.currentSelectedLetter,
+                availableLetters: newState.availableLetters
+            });
+    },
+
+    /**
+     * Get group size.
+     * @param {string} selected letter
+     * @returns {int} group size
+     * @private
+     */
+    _getGroupSize(letter){
+        if(this.state && this.state.data && isArray(this.state.data[letter])){
+            return this.state.data[letter].length;
+        }
+    },
+
+    /**
+     * Render letters filter.
+     * @returns {JSX} Htm content.
+     * @private
+     */
+    _renderLettersFilter(){
+        return (
+            <LettersFilter
+                selectedLetter={this.state.currentSelectedLetter}
+                availableLetters={this.state.availableLetters}
+                handleLetterSelection={this._onLetterSelected}
+                extraCharsSymbol={nonAlphabeticCharactersSymbol}
+                />
+        );
+    },
+
+    /**
+     * Render quick search.
+     * @returns {JSX} Htm content.
+     * @private
+     */
+    _renderQuickSearch(){
+        if(document.querySelector(this.props.externalQuickSearch) == null){
+            return(
+                <div style={{display: 'inline-block'}}>
```

Comment:
that was an experimental use of style prop. I replaced it with a data-focus="..."